frontend/src/components/VolumeBar.tsx

```diff
@@ -0,0 +1,84 @@
+import {useStore} from "../hooks/useStore.ts";
+import styled from "@emotion/styled";
+import {Button} from "@mui/material";
+import SoundIcon from "./icons/SoundIcon.tsx";
+import {useEffect, useState} from "react";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import useSound from "use-sound";
+import click from "../assets/sounds/click.mp3";
+
+export default function VolumeBar() {
+    const [previousVolume, setPreviousVolume] = useState<number>(0)
+    const volume = useStore(state => state.volume)
+    const setVolume = useStore(state => state.setVolume)
+    const [playClick] = useSound(click, {volume: volume});
+
+    useEffect(() => {
+        setPreviousVolume(volume)
+    }, []);
```

Comment:
wont this cause a short flashing of the wrong displayed data on your VolumeBar?

Reply:
So far it didn't.